etc/default/spinnaker

```diff
@@ -12,16 +39,17 @@ SPINNAKER_AWS_DEFAULT_REGION=us-west-2
 # If you plan on using Google Compute Engine set this to true
 SPINNAKER_GOOGLE_ENABLED=false
 # GCP project you intend to manage
-SPINNAKER_GOOGLE_PROJECT_ID=
+SPINNAKER_GOOGLE_PROJECT_ID=`__get_google_metadata_value "projects/project-id"`
```

Comment:
For etc/default/spinnaker, this is problematic as I'd argued offline when we came up with this solution. What will happen is that it will still attempt to get the metadata value, which will fail and return "" as the value. For google, the failure is essentially immediate (the cost of a DNS lookup) because the hostname wont be found (I suppose you could spoof DNS by creating an artificial entry). For AWS, which uses internal IP addresses, it will likely have to time out trying to connect to a server at that address, unless there happens to be a server there, which will likely fail because it wont know the URL. The function will return "" so as not to bind a value. This isnt an issue at this exact moment because I left AWS to default to cassandra so there are no actual AWS call references here. However moving forward, that's what will happen unless there are other environment signals saying we're on AWS that can guard the metadata server attempt.

I have thoughts about some little optimizations that can be done in that file, but am waiting for more than the one call to be introduced.

When not running on GCE, for the time being, the behavior will be to install cassandra unless you run InstallSpinnaker with --noinstall_cassandra.

@duftler, ptal at default/spinnaker one last time. I made a change to how the default bucket is declared. It is ready for merge so you can do that after you look.